### PR TITLE
feat(spans): Record resource spans for clustering

### PR DESCRIPTION
**Pull request by jjbayer**
We previously only recorded span description for [URL clustering](https://develop.sentry.dev/transaction-clustering/) if the span op was `http`. For those spans, the clustering results are not used anymore because we [scrub HTTP spans more radically](https://github.com/getsentry/relay/blob/7844e2bf392e18f8973d1ff4628f29a74e28fcae/relay-event-normalization/src/normalize/span/description/mod.rs#L117) these days.

Record & cluster resource spans instead, because for those, we still keep URL segments under specific circumstances.

Update 2023-11-14: This PR now also removes the clustering rules from project configs entirely, and as an experiment sets the span clustering threshold to 50.

### CODE REVIEW

File: tests/sentry/ingest/test_span_desc_clusterer.py [hunk 1]
```diff
@@ -83,51 +81,6 @@ def test_distribution():
...
-                        "description": description,
-                        "op": op,
-                        "data": {"description.scrubbed": description_scrubbed},
-                    }
-                ]
-            },
-        )
-        assert len(mocked_record.mock_calls) == expected
-
-
```
Comment (jjbayer) [inline tests/sentry/ingest/test_span_desc_clusterer.py:130]: We only look at the scrubbed description for resource spans, so this test case seemed redundant (there is already a `test_record_span_desc_url`).

File: src/sentry/ingest/transaction_clusterer/datasource/redis.py [hunk 2]
```diff
@@ -201,37 +199,28 @@ def record_span_descriptions(
...
-    """Return the URL from the span description.
-
-    It's assumed the description is an HTTP span's description, with the
-    following format: `<http verb> <url>`.
-    """
-    tokens = description.split(" ")
-    if len(tokens) != 2:
-        return None
-    url = tokens[1]
-    return urlparse(url).path
```
Comment (jjbayer) [inline src/sentry/ingest/transaction_clusterer/datasource/redis.py:220]: Resource spans do not have a `GET ` prefix, so this helper can be removed.

**Review by olksdr — APPROVED** [review]

File: src/sentry/ingest/transaction_clusterer/datasource/redis.py [hunk 3]
```diff
@@ -201,37 +199,28 @@ def record_span_descriptions(
 
 
 def _get_span_description_to_store(span: Mapping[str, Any]) -> Optional[str]:
-    if not span.get("op", "").startswith("http"):
+    if not span.get("op") in ("resource.css", "resource.js"):
```
Comment (iker-barriocanal) [inline src/sentry/ingest/transaction_clusterer/datasource/redis.py:202]: Is `resource.js` an op? I don't see it documented in [dev docs](https://develop.sentry.dev/sdk/performance/span-operations/#browser). Related -- should we include more `resource.*` spans?

**Review by iker-barriocanal — APPROVED** [review]

**Comment by jjbayer** [comment]
After some discussion we decided to go with a deterministic approach for now. See https://github.com/getsentry/relay/pull/2654.

**Comment by jjbayer** [comment]
Update: We will not enforce discovered rules in relay, but will use the clusterer to collect patterns that might be converted to manual rules.

**Pull request merged**